File: app/src/main/java/com/hust_twj/zademo/JetPack/life_cycle/LifeCycleActivity.java
package com.hust_twj.zademo.JetPack.life_cycle;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;

import com.hust_twj.zademo.R;

/**
 * Description ：
 * Created by Wenjing.Tang on 2019-08-07.
 */
public class LifeCycleActivity extends AppCompatActivity {

    /**
     *
     * @param savedInstanceState
     */
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_life_cycle);

        findViewById(R.id.tv_lifecycle).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(LifeCycleActivity.this, LifeCycleActivityB.class));
            }
        });

        getLifecycle().addObserver(new MyLifeCycleObserver());
    }

}

File: app/src/main/java/com/hust_twj/zademo/life_cycle/LifeCycleActivity.kt
package com.hust_twj.zademo.life_cycle

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.hust_twj.zademo.R
import kotlinx.android.synthetic.main.activity_lifecycle_main.*


/**
 *
 * Created by Wenjing.Tang on 2019-07-09.
 */
class LifeCycleActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_lifecycle_main)

        tv_activity_life_cycle.setOnClickListener {
            startActivity(Intent(this@LifeCycleActivity, ActivityLifeCycleActivity::class.java))
        }

        tv_service_life_cycle.setOnClickListener {
            startActivity(Intent(this@LifeCycleActivity, ServiceLifeCycleActivity::class.java))
        }

    }

}
